### PR TITLE
docs: simplify Terraform sections and convert to steps.

### PR DESCRIPTION
**Pull request by lsgunn-teleport**
terraform-provider.mdfx
- Moved some information to the main body.
- Changed wording to use second person PoV, present tense, simpler sentence structure.
- Converted paragraphs to steps (because it makes them easier to read).

add-role-to-user.mdx:
- Converted paragraphs to steps (because it makes them easier to read).
- Moved the warning to the more relevant step.
- Add the sign out/sign back in as the file step on each tab (less distracting).

### CODE REVIEW

File: docs/pages/includes/add-role-to-user.mdx [hunk 1]
```diff
@@ -1,147 +1,148 @@
...
-```code
-$ tctl create -f github.yaml
-```
-
-<Admonition type="warning">
-Note the `--with-secrets` flag in the `tctl get` command. This adds the value of
-`spec.signing_key_pair.private_key` to `github.yaml`. This is a sensitive value,
-so take precautions when creating this file and remove it after updating the resource.
-</Admonition>
+1. Retrieve your `github`  configuration resource:
```
Comment (zmb3) [inline docs/pages/includes/add-role-to-user.mdx:35]:
```suggestion
1. Retrieve your `github` Auth Connector:
```

File: docs/pages/management/dynamic-resources/terraform-provider.mdx [hunk 2]
```diff
@@ -22,151 +22,165 @@ This guide will explain how to:
...
-```
-</TabItem>
-</Tabs>
+Terraform needs a signed identity file from the Teleport cluster certificate
+authority to manage resources in the cluster.
+You can create a local Teleport user for this purpose or you can use a 
+[Machine ID](../../machine-id/introduction.mdx) to generate credentials.
+
+If you intend to run Terraform from a CI/CD platform, a Machine ID is often a better 
+option for generating credentials. A Machine ID can provision ephemeral short-lived
```
Comment (zmb3) [inline docs/pages/management/dynamic-resources/terraform-provider.mdx:33]:
```suggestion
option for generating credentials.  Machine ID can provision ephemeral short-lived 
```

Machine ID is one of our product lines, so I think this made more sense before. (We would say "Teleport can X", not "A Teleport can X")

**Review by zmb3 — APPROVED** [review]

File: docs/pages/includes/add-role-to-user.mdx [hunk 3]
```diff
@@ -1,147 +1,148 @@
...
-```code
-$ tctl create -f github.yaml
-```
-
-<Admonition type="warning">
-Note the `--with-secrets` flag in the `tctl get` command. This adds the value of
-`spec.signing_key_pair.private_key` to `github.yaml`. This is a sensitive value,
-so take precautions when creating this file and remove it after updating the resource.
-</Admonition>
+1. Retrieve your `github` Auth Connector:
```
Comment (ptgott) [inline docs/pages/includes/add-role-to-user.mdx:35]:
```suggestion
1. Retrieve your `github` authentication connector:
```

For consistency with the rest of the docs:

```
% grep -hRioE "auth(entication) connector" docs/pages | sort | uniq -c
   1 Authentication Connector
   4 Authentication connector
  36 authentication connector
```

File: docs/pages/management/dynamic-resources/terraform-provider.mdx [hunk 4]
```diff
@@ -1,12 +1,12 @@
 ---
 title: Terraform Provider
-description: Configuration as code with the Teleport Terraform Provider
+description: How to deploy infrastructure-as-code using the Teleport Terraform provider.
```
Comment (ptgott) [inline docs/pages/management/dynamic-resources/terraform-provider.mdx:3]:
```suggestion
description: How to manage dynamic resources as code using the Teleport Terraform provider.
```
For precision

File: docs/pages/management/dynamic-resources/terraform-provider.mdx [hunk 5]
```diff
@@ -22,151 +22,165 @@ This guide will explain how to:
...
-</TabItem>
-<TabItem scope={["oss", "enterprise"]} label="Self-Hosted">
-```
-(!examples/resources/terraform/terraform-user-role-self-hosted.tf!)
-```
-</TabItem>
-</Tabs>
+Terraform needs a signed identity file from the Teleport cluster certificate
+authority to manage resources in the cluster.
+You can create a local Teleport user for this purpose or you can use a
```
Comment (ptgott) [inline docs/pages/management/dynamic-resources/terraform-provider.mdx:29]: The product name is "Machine ID", not "a Machine ID". There are other instances of this as well.

File: docs/pages/management/dynamic-resources/terraform-provider.mdx [hunk 6]
```diff
@@ -22,151 +22,165 @@ This guide will explain how to:
...
+     allow:
+       # This impersonate rule will allow any user with this role to impersonate
+       # and generate certificates for the user named "terraform" with a role also
+       # named "terraform".
+       impersonate:
+         users: ['terraform']
+         roles: ['terraform']
+   ```
+
+1. Create the terraform-impersonator role by running the following command:
```
Comment (ptgott) [inline docs/pages/management/dynamic-resources/terraform-provider.mdx:121]:
```suggestion
1. Create the `terraform-impersonator` role by running the following command:
```

**Review by ptgott — APPROVED** [review]
Approved with suggestions. The big one is that these changes say "a Machine ID", while the name of the product is "Machine ID".

File: docs/pages/management/dynamic-resources/terraform-provider.mdx (hunk 5 above)
Comment (lsgunn-teleport) [reply docs/pages/management/dynamic-resources/terraform-provider.mdx:29]: Personally, I'm not a big fan of this as a formal product name (generic terminology used as if it is a proper noun always strikes me as odd). I also see machine identity agent is sometimes used, so there still might be some inconsistency around the name.

**Pull request merged**